system_setup/client/client.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import sys
+
+from subiquity.client.client import SubiquityClient
+
+log = logging.getLogger('system_setup.client.client')
+
+
+class SystemSetupClient(SubiquityClient):
+
+    from system_setup.client import controllers as controllers_mod
+
+    snapd_socket_path = None
+
+    variant = "wsl_setup"
+
+    controllers = [
+        "Welcome",
+        "WSLIdentity",
+        "WSLConfigurationBase",
+        "Summary",
+        ]
+
+    def __init__(self, opts):
+        # TODO WSL:
+        # 1. remove reconfigure flag
+        # 2. decide on which UI to show up based on existing user UID >=1000
+        #    (or default user set in wsl.conf?)
+        # 3. provide an API for this for the flutter UI to know about it
+        # 4. Add Configuration Base page before Advanced
+        # 5. Add language page
+        # self.variant = "wsl_configuration"
+        if opts.reconfigure:
+            self.controllers = [
+                "WSLConfigurationBase",
+                "WSLConfigurationAdvanced",
+                "Summary",
+            ]
```

Comment:
In general my approach for conditional screens has been for the client to be prepared to show all of them and the server to tell the client not to show a particular screen by sending an x-override-status: skip header (this is 100% not something you need to fix before merge!)

Reply:
There is a TODO to finish the implementation of the  reconfiguration mode. The change you request will be part of this work.